Kamelet API test rate limit

```
--- FAIL: TestNewRepositoryWithCamelKamelets (0.06s)
    repository_test.go:151: 
        	Error Trace:	repository_test.go:151
        	Error:      	Received unexpected error:
        	            	GET https://api.github.com/repos/apache/camel-kamelets/contents/kamelets: 403 API rate limit exceeded for 20.230.23.20. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.) [rate reset in 50m05s]
        	Test:       	TestNewRepositoryWithCamelKamelets
    repository_test.go:152: 
        	Error Trace:	repository_test.go:152
        	Error:      	Should be true
        	Test:       	TestNewRepositoryWithCamelKamelets
```
I think we need to review that test to mitigate the number of API hit.

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -17,6 +17,9 @@
 
 name: build
 
+env:
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
 on:
   pull_request:
     branches:

--- a/.github/workflows/builder.yml
+++ b/.github/workflows/builder.yml
@@ -17,6 +17,9 @@
 
 name: builder
 
+env:
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
 on:
   pull_request:
     branches:

--- a/.github/workflows/common.yml
+++ b/.github/workflows/common.yml
@@ -17,6 +17,9 @@
 
 name: common
 
+env:
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
 on:
   pull_request:
     branches:

--- a/.github/workflows/install.yml
+++ b/.github/workflows/install.yml
@@ -17,6 +17,9 @@
 
 name: install
 
+env:
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
 on:
   pull_request:
     branches:

--- a/.github/workflows/knative.yml
+++ b/.github/workflows/knative.yml
@@ -17,6 +17,9 @@
 
 name: knative
 
+env:
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
 on:
   pull_request:
     branches:

--- a/.github/workflows/local.yml
+++ b/.github/workflows/local.yml
@@ -17,6 +17,9 @@
 
 name: local
 
+env:
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
 on:
   pull_request:
     branches:

--- a/.github/workflows/openshift.yml
+++ b/.github/workflows/openshift.yml
@@ -17,6 +17,9 @@
 
 name: openshift
 
+env:
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
 on:
   pull_request:
     branches:

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -17,6 +17,9 @@
 # limitations under the License.
 name: release
 
+env:
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
 on:
   schedule:
     - cron: "10 23 * * *"

--- a/.github/workflows/upgrade.yml
+++ b/.github/workflows/upgrade.yml
@@ -17,6 +17,9 @@
 
 name: upgrade
 
+env:
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
 on:
   pull_request:
     branches:

